### PR TITLE
Refactor markup code block styles

### LINKED ISSUE #30292: Code markup style regression
The inner background is new. Don't know if that is intended but here it looks strange:
grafik

### PR DESCRIPTION
**Pull request by wxiaoguang**
Close #30292

image

image

### CODE REVIEW

**Comment by silverwind** [comment]
I did a different fix for this in https://github.com/go-gitea/gitea/pull/30282:

```css
.markup .highlight pre > code,
.markup pre > code {
  background: none !important;
}
```

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
@silverwind I believe is good enough and it is a right fix. It clearly defines the styles for different `code` usages, and remove all dirty/unclear CSS style overridings, and no need to use `!important`.

File: web_src/css/markup/content.css [hunk 1]
```diff
@@ -451,15 +444,16 @@
...
+.markup p > code,
+.markup li > code,
+.markup p > tt,
+.markup li > tt {
+  padding: 0.2em 0.4em;
+  font-size: 85%;
+  white-space: break-spaces;
+  background-color: var(--color-markup-code-block);
+  border-radius: var(--border-radius);
 }
```
Comment (silverwind) [inline web_src/css/markup/content.css:457]: I disagree with this selector list. what about `blockquote > code` etc, you can't possibly list every single element here.

**Review by silverwind — CHANGES_REQUESTED** [review]
Would like to see `.markup code:not(.code-inner)` being reverted to `.markup code`. This is too hacky.

**Comment by wxiaoguang** [comment]
I don't see anything hacky. The existing code is more hacky and will cause more problems.

**Comment by silverwind** [comment]
If we really have to we, can use `.markup code:where(:not(.code-inner))` to not add specificity in a `:not` selector but I would prefer to stay as close as possible to GitHub CSS and that is via `.markup code`.

**Comment by wxiaoguang** [comment]
I have no interesting for playing style overriding games, as always.

**Comment by wxiaoguang** [comment]
->  Fix code block style for code preview #30298 

I think it is better than overriding unnecessary styles.

**Pull request closed without merging**